qdrant_client/local/order_by.py

```diff
@@ -0,0 +1,52 @@
+from datetime import datetime
+from typing import Any, Optional, Union
+
+from dateutil.parser import parse
+
+MICROS_PER_SECOND = 1_000_000
+
+
+class OrderingValue:
+    def __init__(self, value: Union[int, float]):
+        self.value = value
+
+    def __lt__(self, other: "OrderingValue") -> bool:
+        return self.value < other.value
+
+    def __gt__(self, other: "OrderingValue") -> bool:
+        return self.value > other.value
+
+    def __ge__(self, other: "OrderingValue") -> bool:
+        return self.value >= other.value
+
+    def __le__(self, other: "OrderingValue") -> bool:
+        return self.value <= other.value
+
```

Comment:
Why do we need this structure?

Reply:
I feel like it makes types simpler, instead of applying cmp operators on a Union + converting datetimes to a specific type

Reply:
We don't actually need this, switching to type alias should be fine